src/main/java/io/github/hotlava03/chatutils/fileio/ChatStorage.java

```diff
@@ -0,0 +1,97 @@
+package io.github.hotlava03.chatutils.fileio;
+
+import com.google.gson.*;
+import io.github.hotlava03.chatutils.util.IoUtils;
+import org.apache.logging.log4j.LogManager;
+
+import java.io.*;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+public class ChatStorage {
+    public static final int MAX_ENTRIES = 100;
+    public static final String SINGLEPLAYER_ADDRESS = "minecraft.singleplayer";
+    private static ChatStorage instance;
+    private final Gson gson = new Gson();
+    private JsonObject object = new JsonObject();
+    private boolean lockingChatEvents = true;
```

Comment:
honestly this made me think of mutexes and semaphores where locks and keys are held by functions, it's probably more appropriate to call it `blockingChatEvents` (and getters/setters accordingly)